### PR TITLE
refactor(sdk): typescript update

### PR DESCRIPTION
**Pull request by DamienLaf**
## 📝 What is this PR

-   [Specs](https://phantombuster.atlassian.net/browse/BE-204)

This is a rewriting of the current `phantombuster-sdk` in typescript.

## 🗒 Notes

To keep the behavior unchanged, tests have been made on the previous code.

here is the coverage:
image

-   [ ] This change requires a documentation update
-   [x] This change requires a unit tests update

### CODE REVIEW

**Review by ghermet — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,45 +1,77 @@
...
-    "needle": "0.10.x",
-    "node-watch": "^0.6.2",
-    "yargs": "3.27.x"
-  }
+	"name": "phantombuster-sdk",
+	"version": "0.4.0",
+	"description": "Phantombuster's SDK",
+	"main": "build/sdk.js",
+	"scripts": {
+		"test": "echo \"Error: no tests exist\" && exit 1",
```
Comment (Drax35c2db) [inline package.json]: Please update the `test` command

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import type { AsTyped } from "as-typed"
+import type conf from "./configuration.schema"
+
+export type ConfType = Array<
+	AsTyped<typeof conf>[number] & {
+		scripts?: Record<string, string>
+	}
+>
```
Comment (Drax35c2db) [inline src/types.ts:9]: Can we have a comment on the necessity of the record string string cast plz ? :)

File: package.json [hunk 3]
```diff
@@ -1,45 +1,77 @@
...
+		"@types/jest": "^29.5.12",
+		"@types/needle": "^3.3.0",
+		"@types/node": "^22.5.2",
+		"@typescript-eslint/eslint-plugin": "^8.4.0",
+		"@typescript-eslint/parser": "^8.4.0",
+		"as-typed": "^1.3.2",
+		"eslint": "^8.57.0",
+		"eslint-config-prettier": "^9.1.0",
+		"eslint-plugin-deprecation": "^3.0.0",
+		"eslint-plugin-diff": "^2.0.3",
```
Comment (Drax35c2db) [inline package.json]: remove ?

File: src/constants.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const defaultEndpoint = "https://phantombuster.com/api/v1"
+
+export const extentionRegexp = /.[^.]+$/
```
Comment (Drax35c2db) [inline src/constants.ts]:
```suggestion
export const extensionRegexp = /\.[^.]+$/
```

File: src/sdk.test.ts [hunk 5]
```diff
@@ -0,0 +1,705 @@
...
+				json: '{\n\t"slug": "test"\n}\n',
+				md: "markdown test\n",
+			})
+		})
+	})
+
+	describe("script file change", () => {
+		const file = "src/__fixtures__/test.js"
+
+		it("should not log if script does not exists bis", async () => {
```
Comment (Drax35c2db) [inline src/sdk.test.ts]:
```suggestion
		it("should not log if script does not exists", async () => {
```

File: src/Logger.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+function datePrefix() {
+	return new Date().toLocaleTimeString() + " - "
+}
+
+export const Logger = {
+	log(...[firstParams, ...params]: unknown[]) {
+		console.log(`${datePrefix()}${firstParams}`, ...params)
+	},
+	success({ account, scriptName, scriptPath, realPath }: IScriptConfig, created = false) {
+		const label = realPath.endsWith(".json") || realPath.endsWith(".md") ? "[API store settings] " : ""
```
Comment (Drax35c2db) [inline src/Logger.ts]: Maybe Put json and md logic outside of here ?

**Review by Drax35c2db — CHANGES_REQUESTED** [review]

File: package.json [hunk 7]
```diff
@@ -1,45 +1,76 @@
...
-  },
-  "dependencies": {
-    "cson": "3.0.x",
-    "is-my-json-valid": "2.12.x",
-    "needle": "0.10.x",
-    "node-watch": "^0.6.2",
-    "yargs": "3.27.x"
-  }
+	"name": "phantombuster-sdk",
+	"version": "0.4.0",
```
Comment (Hatscat) [inline package.json]: Even if there is no behavior change, the deps are different, the built code is different and the package "scripts" are different. WDYT?
```suggestion
	"version": "0.4.1",
```

Comment (DamienLaf) [reply package.json]: I think it should be the v1.0, but I wait for @Drax35c2db to tell me how I should bump the version and upgrade the package

**Review by Drax35c2db — APPROVED** [review]
LGTM

> I think it should be the v1.0, but I wait for @Drax35c2db to tell me how I should bump the version and upgrade the package

Let's validate this version with @paps first and then we can make more official next steps like npm versions bump

**Pull request merged**